build-logic/src/main/kotlin/publishing/PublishingExtension.kt

```diff
@@ -0,0 +1,64 @@
+package publishing
+
+import com.vanniktech.maven.publish.SonatypeHost
+import gradle.kotlin.dsl.accessors._089c671483fb7c20aeca81bf72df85c9.mavenPublishing
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.provideDelegate
+
+private const val ElmslieGitHubUrl = "https://github.com/vivid-money/elmslie"
+
+abstract class PublishingExtension(
+    private val project: Project,
+) {
+
+    fun configure(
+        pom: Pom,
+    ) {
+        val libraryGroup: String by project
+        val libraryVersion: String by project
+
+        with(project.mavenPublishing) {
```

Comment:
all this code must put be inside convention plugin with only pom.name and pom.description being configurable via this extension